plothist/plotters.py

```diff
@@ -512,18 +516,26 @@ def plot_comparison(
     ax.errorbar(
         x=hist_2.axes[0].centers,
         xerr=None,
-        y=np.nan_to_num(ratio_values, nan=0),
-        yerr=np.nan_to_num(np.sqrt(ratio_variance), nan=0),
+        y=np.nan_to_num(comparison_values, nan=0),
+        yerr=np.nan_to_num(np.sqrt(ratio_variance), nan=0)
+        if (ratio_uncertainty == "uncorrelated" or comparison == "pull")
+        else np.nan_to_num(h1_scaled_uncertainty, nan=0),
         fmt=".",
         color="black",
     )
 
     if comparison == "ratio":
+        if comparison_ylim is None:
+            comparison_ylim = (0.0, 2.0)
+        ax.set_ylim(comparison_ylim)
```

Comment:
This should be done only once at the end of the function, together with `set_xlim()`